staging/src/k8s.io/kube-scheduler/config/v1alpha1/types.go

```diff
@@ -127,3 +136,64 @@ type KubeSchedulerLeaderElectionConfiguration struct {
 	// LockObjectName defines the lock object name
 	LockObjectName string `json:"lockObjectName"`
 }
+
+// Plugins include multiple extension points. When specified, the list of plugins for
+// a particular extension point are the only ones enabled. If an extension point is
+// omitted from the config, then the default set of plugins is used for that extension point.
+// If no plugin should be invoked at an extension point, an empty list must be provided.
+type Plugins struct {
+	// QueueSort is a list of plugins that should be invoked when sorting pods in the scheduling queue.
+	QueueSort []Plugin `json:"queueSort,omitempty"`
+
+	// PreFilter is a list of plugins that should be invoked at "PreFilter" extension point of the scheduling framework.
+	PreFilter []Plugin `json:"preFilter,omitempty"`
+
+	// Filter is a list of plugins that should be invoked when filtering out nodes that cannot run the Pod.
+	Filter []Plugin `json:"filter,omitempty"`
+
+	// PostFilter is a list of plugins that are invoked after filtering out infeasible nodes.
+	PostFilter []Plugin `json:"postFilter,omitempty"`
+
+	// Score is a list of plugins that should be invoked when ranking nodes that have passed the filtering phase.
+	Score []Plugin `json:"score,omitempty"`
+
+	// NormalizeScore is a list of plugins that should be invoked after the scoring phase to normalize scores.
+	NormalizeScore []Plugin `json:"normalizeScore,omitempty"`
+
+	// Reserve is a list of plugins invoked when reserving a node to run the pod.
+	Reserve []Plugin `json:"reserve,omitempty"`
+
+	// Permit is a list of plugins that control binding of a Pod. These plugins can prevent or delay binding of a Pod.
+	Permit []Plugin `json:"permit,omitempty"`
+
+	// PreBind is a list of plugins that should be invoked before a pod is bound.
+	PreBind []Plugin `json:"preBind,omitempty"`
+
+	// Bind is a list of plugins that should be invoked at "Bind" extension point of the scheduling framework.
+	// The scheduler call these plugins in order. Scheduler skips the rest of these plugins as soon as one returns success.
+	Bind []Plugin `json:"bind,omitempty"`
+
+	// PostBind is a list of plugins that should be invoked after a pod is successfully bound.
+	PostBind []Plugin `json:"postBind,omitempty"`
+
+	// UnReserve is a list of plugins invoked when a pod that was previously reserved is rejected in a later phase.
+	UnReserve []Plugin `json:"unReserve,omitempty"`
```

Comment:
is "unReserve" used elsewhere already, or is "unreserve" a possibility?

Reply:
Good catch. We should go with "Unreserve" in Go and "unreserve" in json.